### ISSUE TITLE
[FEATURE REQUEST] update Discover in Fen Light to match FEN including autocomplete. Add dates filter. Work on updating anime section 

### ISSUE THREAD

**Issue by hcgiub001** [Tikipeter/tikipeter.github.io#151]
Anime is Japanese Animation. Instead of searching for keyword anime. I've searched for 
Language: Japanese
Genre:Animation

this is 1:1 with using Anime as a keyword.

If you use more than one keyword in TMDB website search, it's an OR operator and not an AND operator.

this is a subgenre of anime
Shoujo: a genre of Japanese comics and animated films aimed primarily at a young female audience, typically characterized by a focus on personal and romantic relationships.

If we search using two keywords, "anime" & "Shoujo"
we get this result
![image](https://github.com/user-attachments/assets/501426a5-5b60-4cb8-b9c0-98d53bb6a47c)

looking at first anime
[Dragon Ball DAIMA](https://www.themoviedb.org/tv/236994-daima?language=en-GB) (2024)
this is the keywords it has
![image](https://github.com/user-attachments/assets/8e5f2e0f-46af-458e-b400-1cfe68dfe00e)

this isn't what I wanted I wanted to search for Shoujo.
Now instead of anime, if I use language Japanese and genre animation and use the keyword Shoujo , I get this
![image](https://github.com/user-attachments/assets/1d341404-6fb0-4b5a-8855-bdb384e66021)

BINGO. Do you know how powerful this is? all this time we kept saying TMDB doesn't have anime genre. They could easily create a section just using discover filters. Of course not every anime will have the right keyword but I'm confident this could beat SIMKL or even OTAKU.

Look at this for "military" subgenre of anime using this method
![image](https://github.com/user-attachments/assets/88b311f5-e936-442e-96d3-399606f868e8)

This is amazing, and not only that having keywords is way more powerful than even OTAKU. Now I'd like to take on a project with your help (if you are willing)

I want to create a list of genres similar to say OTAKU
100% we'll get every genre possible with using keywords. BUT I want to do one better. So I'll have a list of genres

![image](https://github.com/user-attachments/assets/695345aa-621e-45af-a23a-26e6f151088b)

but each genre will be a folder, so inside action genre we'll have search term for just action at the top but
underneath we'll have subfolders with the subgenres of action (Still need to make a list of what those will be)

TMDB is so powerful. this will be better than any genre list out their. 
But I've realised your discover feature in Fen Light is a strip down version of the one in FEN.
right now I can accomplish all this in FEN .

Realising how powerful discover is (you was right before) we can update the discover in Fen Light?
If so I want some more filters. one of them is similar to
![image](https://github.com/user-attachments/assets/405cbda6-b9e3-4a5b-bcc9-68609d7e01f5)

So we can get anime that are airing on a specific date. I want the ability to go back or forward from present day. so effectively I want to have a list similar to Haro with airing anime
![image](https://github.com/user-attachments/assets/647a01d6-fb58-4c0c-8214-333585db5a48)

TMDB is working perfectly for this already. I can search for anime that is airing today. 
so now we are building a really good anime section.

So I can handle all the listings , you can let me know what you want from me, like after I done all the folders with genres and everything , you want the file?

This goes beyond Anime. nothing stopping us creating a folder of a language, say French and then list all genres of french. Or Chinese animation . Or korean Dramas. I want to work on anime and see how well it will turn out and then have a go at kdrama

Also the keywords are so amazing way to search that I wanted an  autocomplete field, and luck was on my side
"Daily ID export is now live."
![image](https://github.com/user-attachments/assets/72228400-2911-456a-8437-674ef30251c0)
https://www.themoviedb.org/talk/588418309251410457012d57

https://developer.themoviedb.org/docs/daily-id-exports

using
http://files.tmdb.org/p/exports/keyword_ids_MM_DD_YYYY.json.gz

I can get a list of all the keywords. I'm not sure how you will do this but is their a way to have an autocomplete function for keywords using this list?

The discover is so amazing that I had no clue
Thanks for reading this
TLDR I'd like to work on creating a structure to anime section to literally make it a complete anime addon that can stand on its own 
I can go ahead and start working on the structure and use FEN to create it, then I'll have the file that I can send to you to look at 
I'd like Fen Light to have a more Discover feature equal and better than FEN has with dates adition so I can search for today's date, and 1 day before, 2 day before etc. basically I want to have a list of all days of the week like Haru has
I'd like an autocomplete feature for keywords. Also is their a way to have multiple keywords using AND operator, unlike the tmdb website that is using OR operator.

Kind of excited what the possibilities are. Tahnkyou for taking time to read, if you have any thoughts or questions , I'm happy to hear them

**Comment by hcgiub001** [comment]
I could just have a list of genres similar to how the industry uses them, so a list similar to what I posted above. But TMDB has it's own strengths and weakness's and quirky choices for keywords.
I wanted your advice if it's okay to scrape all the shows with the keyword "anime" from tmdb, then to get a list of all the keywords for each anime and have a tally of the words TMDB likes to use.
Using this we could not only have industry standard genres but have genres that are TMDB specific.
This could also help with auto-complete , with knowing what keywords are used you can only have them come up.

I've been playing with Anime section and I don't think people understand the full capability of what Fen Light can do with anime with it's integration with TMDB, Trakt and IMDB recommendations.

I've come across a show then use context menu to get IMDB recommendations (way more accurate than TMDB reccomendations), then I checked what Trakt list's it's included in, and found amazing lists that are representative of the anime I'm looking at. We have trailers, reviews, cast where we can find other works by them.

Now with a bigger and more complete genre list, you can find so much anime with little effort, once people use Fen Light to it's maximum.
Is it possible to get all the keywords for anime and have a spreadsheet with all a tally of them?

**Comment by hcgiub001** [comment]
Sorry had to post this because I think it's amazing development. I had no idea we can edit TMDB pages
Here is a list of Japanese animation with keyword horror
![image](https://github.com/user-attachments/assets/11322b52-7833-48ed-9eef-5ca5b18690a9)
15 entries

here is a list from MyAnimeList
![image](https://github.com/user-attachments/assets/8c9f6827-1490-4824-9ae2-5b62557fe2d5)

All someone has to do is spend a few hours to tag horror to those anime and we have parity with the best anime site. If you go ahead and add this feature request , I'm going to create the genre list, then once this is live, I'll ask for help from addons4kodi community that use Fen Light for anime to pick a genre they like and to find each TMDB id for that genre and tag the correct keyword. In a week we could literally have as good anime section in Fen Light as Otaku. I'm curious how long it will take me to tag the horror, so I'm having a go at that now.

I'm also looking into adding Networks section (or studios as it's called in MAL) in anime section.
So users can look through studios to find their anime. like Crunchyroll etc.

This really will end up being a true anime addon. and you have episode groups coming so that's another 99% of anime we'll get, especially all the new stuff.
